### PR TITLE
sp-dh-new-shopping-list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items.
A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following sequence:

* Generate a new, unique token
* Save the token to `localStorage`
* Show the user the list view

The script `getToken` from the `shopping-list-utils` package should be used to generate the token. More information on that package [here](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils).

`import { getToken, words } from '@the-collab-lab/shopping-list-utils'`

AC:

* `@the-collab-lab/shopping-list-utils` has been added as a dependency to the project.
* For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
* Clicking the button/link generates a new token and saves it to `localStorage`
* Once the token has been created and saved, the user is redirected to the "list" view
* For users who do have a token already saved in `localStorage`, they should be shown/redirected to the "list" view

Notes:

* Eventually the welcome/home page will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
* It's good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
* For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by sparsell**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

The code in this PR aims to do the following:
 - Upon loading, the Home component checks to see if localStorage contains a key named 'token' associated with the User. 
 - If there is a token, the JoinList component is rendered:

-- Once rendered,

    -- The value of the existing token will be shown as the default value in the input field
    -- Render the "List" view

- If there isn't a token, the CreateList component is rendered:

-- Once rendered, a button is clicked to:

    -- Generate a new token
    -- Save the new token to localStorage
    -- Renders the "List" view

We created two components (CreateList and JoinList) in order to simplify the Home component's return function as well as making them easy to reuse if necessary. 

## Related Issue

closes #3

## Acceptance Criteria

- @the-collab-lab/shopping-list-utils has been added as a dependency to the project.
- For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
- Clicking the button/link generates a new token and saves it to localStorage
- Once the token has been created and saved, the user is redirected to the "list" view
- For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### After

 - CreateList (User does not yet have a token)
Screen Shot 2022-04-13 at 7 59 08 PM

- JoinList (if User already has a token)

Screen Shot 2022-04-13 at 9 16 01 PM

## Testing Steps / QA Criteria

In the App:

- Pull down the changes from the branch
`git pull`

- Checkout to our branch
`git checkout sj-dh-new-shopping-list`

- Run server
`npm start`

- Open `localhost:3000` and open your console

- Navigate to the "Application" tab and on the left under the Storage heading, click on "Local Storage". 
Screenshot 2022-04-13 201434

- There should be nothing under "Key" and "Value" (ignore the "ally-supports-cache" in the screenshot above)

- Click on the "Join an existing list" button while the Application window is still open

- After you click on the button there should be a new Key of "token" and a Value of 3 randomly generated words in the Application window
Animation

- Click on the "Home" link. You should then see an input field that has already been filled out with the same 3 words from the Value window
Animation1

- If you want to retest, right click on the token inside your Local Storage and click "Delete" and then refresh and click on the button to generate a new token.
Animation2

### CODE REVIEW

**Review by k-schrattenholzer — APPROVED** [review]
I was able to pull down your branch, and see the changes! When I clicked the Join List button on the Create List view, it generated a token in Local Storage and when I returned to the Home view that token was reflected in the input. 
The only note I had was around the display text on the Create List Button. Right now it says Join an Existing List - which was a little confusing in testing. Can we update the text to say "Create list" or something similar?

**Review by artsycoder533 — APPROVED** [review]
When I get to the home screen I see a join an existing list button, when I click this button I am directed to the list view.  When I navigate back to the home screen I now see an input box containing the token.  I checked local storage and i see a key label "token" with a value that corresponds to the text inside the input field. The only thing I noticed is the CreateList view has a button with the text "Join and existing list" So the conditional in the Home component renders the same button essentially.  Double check the text of button, is it supposed to say "Join a new list" or "Create a new list"?  All in all it works great!

File: src/views/Home/Home.jsx [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+  const [hasToken, setHasToken] = useState(token !== null);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    setHasToken(token !== null);
+  }, []);
+
+  const handleCreateToken = () => {
+    setToken(localStorage.setItem('token', getToken()));
+  };
```
Comment (mxmason) [inline src/views/Home/Home.jsx:17]: While the user-facing *behavior* of the feature is correct, this code might not be doing what you expect!

On line 7, we're saying "set the `token` state to whatever is stored at the 'token' key in localStorage". On line 16, when the user clicks the `Create a list` button, we say "set the `token` state to _the  value of_ `localStorage.setItem`. This is what allows the feature to technically function correctly, but I have some food for thought:

- What is the value of calling `localStorage.setItem`? You can debug that in your console. Is it your token? If it isn't, how could you make the code more accurate?

This exercise might not be relevant as you continue building features – you may replace this code altogether, and that's fine! – but it's important to verify that your code is behaving correctly as you go!

**Review by mxmason — APPROVED** [review]
Great work on this! I've left a comment on a specific bit of code for you to consider, but it's not a blocking issue!

**Pull request merged**